tests/Type/GeneralTypeTest.php

```diff
@@ -63,13 +72,20 @@ public static function dataFileAsserts(): iterable
         yield from self::gatherAssertTypes(__DIR__ . '/data/view-exists.php');
         yield from self::gatherAssertTypes(__DIR__ . '/data/view.php');
         yield from self::gatherAssertTypes(__DIR__ . '/data/where-relation.php');
-        yield from self::gatherAssertTypes(__DIR__ . '/data/bug-1997.php');
-        yield from self::gatherAssertTypes(__DIR__ . '/data/bug-1819.php');
 
-        if (version_compare(LARAVEL_VERSION, '11.28.0', '>=')) {
+        if (laravel_version_compare('11.28.0', '>=')) {
             yield from self::gatherAssertTypes(__DIR__ . '/data/model-collections-l11-28.php');
         }
 
+        // 11.100 is an artificial constraint until 12.0.0 is released
+        if (laravel_version_compare('11.0.0', '>=') && laravel_version_compare('11.100.0', '<')) {
+            yield from self::gatherAssertTypes(__DIR__ . '/data/collection-generic-static-methods-l11.php');
+        }
+
+        if (laravel_version_compare('12.0.0', '>=') || LARAVEL_VERSION === '12.x-dev') {
```

Comment:
```suggestion
        if (laravel_version_compare('12.0.0', '>=')) {
```

Testbench Core already check for `12.x-dev`: https://github.com/orchestral/testbench-core/blob/c3d080c533cff1960aa0468dd930cce76fdfea5e/src/functions.php#L417-L430

Reply:
Ah sweet! must've been looking at the wrong version